components/brave_wallet/browser/siwe_message_parser.cc

```diff
@@ -0,0 +1,402 @@
+        }
+        state_ = State::kVersion;
+        break;
+      case State::kVersion:
+        if (!CheckField(token, kVersionToken)) {
+          return {};
+        }
+        if (!base::StringToUint(token.substr(strlen(kVersionToken)),
+                                &result->version)) {
+          return {};
+        }
+        // The only supported version should be 1
+        if (result->version != 1 || !ConsumeDelim(&tokenizer)) {
+          return {};
+        }
+        state_ = State::kChainId;
+        break;
+      case State::kChainId: {
+        std::string chain_id_str;
+        if (!FillStringField(&tokenizer, token, kChainIdToken, true,
+                             chain_id_str)) {
+          return {};
+        }
+        if (!base::StringToUint64(chain_id_str, &result->chain_id)) {
+          return {};
+        }
+        state_ = State::kNonce;
+        break;
+      }
+      case State::kNonce:
+        if (!FillStringField(&tokenizer, token, kNonceToken, true,
+                             result->nonce)) {
+          return {};
+        }
+        if (!base::ranges::all_of(result->nonce,
+                                  &base::IsAsciiAlphaNumeric<char>)) {
+          return {};
+        }
+        if (result->nonce.size() < kMinNonceLength) {
+          return {};
+        }
+        state_ = State::kIssuedAt;
+        break;
+      case State::kIssuedAt:
+        // IssuedAt is the last required field so we don't consume delimiter
+        if (!FillStringField(&tokenizer, token, kIssuedAtToken, false,
+                             result->issued_at)) {
+          return {};
+        }
+        state_ = State::kOptionalFields;
+        seeking_optional_fields = true;
+        break;
+      case State::kOptionalFields: {
+        if (token.empty()) {
+          state_ = State::kEnd;
+          seeking_optional_fields = false;
+          break;
+        }
+        // check and consume starting delimeter
+        if (!tokenizer.token_is_delim() || !tokenizer.GetNext()) {
+          return {};
+        }
+        bool has_expiration_time = FillOptionalStringField(
+            &tokenizer, kExpirationTimeToken, result->expiration_time);
+        // check order of first optional field
+        if (has_expiration_time &&
+            (result->not_before || result->request_id || result->resources)) {
+          return {};
+        }
+        bool has_not_before = FillOptionalStringField(
+            &tokenizer, kNotBeforeToken, result->not_before);
+        // check order of second optional field
+        if (has_not_before && (result->request_id || result->resources)) {
+          return {};
+        }
+        bool has_request_id = FillOptionalStringField(
+            &tokenizer, kRequestIdToken, result->request_id);
+        if (result->request_id &&
+            !base::ranges::all_of(*result->request_id, &IsPChar)) {
+          return {};
+        }
+        // check order of third optional field
```

Comment:
Can we double check the logic of this parsing of resources here? When I was dynamically testing it on MM test dapp I was seeing that this was returning objects to the UI instead of strings which was causing the data to not be displayed properly.

See below:

![Screenshot 2023-09-06 at 8 21 22 PM](https://github.com/brave/brave-core/assets/23125059/00edd1ce-53ff-4f24-b3d8-6ac88ce5249d)

Reply:
can you show me the raw message dapp sends?

Reply:
That is message WITHOUT any resources and core handles it correctly. cc @Douglashdaniel that resources is OPTIONAL, you can test it with https://login.xyz/.

Reply:
updated test plan to check resources specifically.